In [1]:
# suppress warnings
%pip install stata_setup --quiet
import stata_setup
import sys
sys.path.append(r'C:/Program Files/Stata18/utilities')
from pystata import config
config.init('mp')

Note: you may need to restart the kernel to use updated packages.

  ___  ____  ____  ____  ____ ®
 /__    /   ____/   /   ____/      18.0
___/   /   /___/   /   /___/       MP—Parallel Edition

 Statistics and Data Science       Copyright 1985-2023 StataCorp LLC
                                   StataCorp
                                   4905 Lakeway Drive
                                   College Station, Texas 77845 USA
                                   800-STATA-PC        https://www.stata.com
                                   979-696-4600        stata@stata.com

Stata license: Unlimited-user 4-core , expiring 31 May 2024
Serial number: 501809302753
  Licensed to: IFPRI
               IFPRI

Notes:
      1. Unicode is supported; see help unicode_advice.
      2. More than 2 billion observations are allowed; see help obs_advice.
      3. Maximum number of variables is set to 5,000 but can be increased;
          see help set_maxvar.


In [79]:
# import all files in the folder
import glob
import os
import pandas as pd
os.chdir(r'C:\Users\WeilunShi\Dropbox (IFPRI)\FAW\Data\UPDATED-UG-CAP-AFRICA Baseline2019')
extension = 'dta'
all_filenames = [i for i in glob.glob('*.{}'.format(extension))]

# import all files in the folder using df1, df2, ...
for i in range(len(all_filenames)):
    globals()['df{}'.format(i+1)] = pd.read_stata(all_filenames[i])
    globals()['dfv{}'.format(i+1)] = pd.read_stata(all_filenames[i], iterator=True)
    
# rename A1 as subcounty, A2 as parish, A3 as village, A4 as county, A5 as name, A6 as sex, A7 as name_head, A8 as sex_head, A4_1 as head_living_spouse, A9 as relationship, A11 as cell_head, A12 as cell_spouse, A13 as date, A14 as time, A15 as name_enumerator, A16 as name_supervisor
df1 = df1.rename(columns={
    'A1': 'subcounty',
    'A2': 'parish',
    'A3': 'village',
    'A4': 'county',
    'A5': 'name',
    'A6': 'sex',
    'A7': 'name_head',
    'A8': 'sex_head',
    'A4_1': 'head_living_spouse',
    'A9': 'relationship',
    'A11': 'cell_head',
    'A12': 'cell_spouse',
    'A13': 'date',
    'A14': 'time',
    'A15': 'name_enumerator',
    'A16': 'name_supervisor'
})
# drop A9_OTHER
df1 = df1.drop(columns=['A9_OTHER'])
df2 = df2.rename(columns={
    'M2A_0': 'name',
    'M2A_1': 'relationship',
    'M2A_1_OTHER': 'relationship_other',
    'M2A_2':'marital_status',
    'M2A_3': 'sex',
    'M2A4_1': 'attend_school',
    'M2A4_2': 'education',
    'M2A5_1': 'usual_residence',
    'M2A5_2': 'guest',
    'M2A5_3': 'stay_last_night',
    'M2A_6': 'age',
    'MA_7': 'occupation',
    'MA7_1': 'occupation_other',
    'M2A8': 'contribution_farming_activity',
    'ADD_MEMBER': 'add_member_1',
})
# Overlay add_member_1 to add_member, overwrite any existing values

df2['add_member'] = df2['add_member'].astype('object').fillna(df2['add_member_1']).astype('object')
# drop add_member_1
df2 = df2.drop(columns=['add_member_1'])
# uppercase all values in add_member
df2['add_member'] = df2['add_member'].str.upper()
df3 = df3.rename(columns={
    'HF04': 'HH_crop_1',
    'HF041': 'HH_crop_2',
    'HF042': 'HH_crop_3',
    'HF043': 'HH_crop_4',
    'HF044': 'HH_crop_5',
    'HF04_OTHER': 'HH_crop_other',
    'M31C_JUL_AUG_2019': 'total_land_cultivated_jul_aug_2019',
    'M31C_FEB_MAR_2019': 'total_land_cultivated_feb_mar_2019',
    'M31C_AUG_SEPT_2018': 'total_land_cultivated_aug_sept_2018',
    'M31D_JUL_AUG_2019': 'total_land_damaged_jul_aug_2019',
    'M31D_FEB_MAR_2019': 'total_land_damaged_feb_mar_2019',
    'M31DAUG_SEPT_2018': 'total_land_damaged_aug_sept_2018',
    'M31E_I': 'FAW_awareness',
    'M31E_II': 'Recognize_FAW',
    'M31E_III': 'FAW_major_constraint',
    'M31F_I':'stem_borer_awareness',
    'M31F_II': 'Recognize_stem_borer',
    'M31F_III': 'stem_borer_major_constraint',
    'M31G':'striga_constraint',
    'M31H':'striga_seriousness',
    'M31I_JUL_AUG_2019': 'maize_area_jul_aug_2019',
    'M31I_FEB_MAR_2019': 'maize_area_feb_mar_2019',
    'M31I_AUG_SEPT_2018': 'maize_area_aug_sept_2018',
    'M31J_JUL_AUG_2019': 'maize_yield_jul_aug_2019',
    'M31J_FEB_MAR_2019': 'maize_yield_feb_mar_2019',
    'M31J_AUG_SEPT_2018': 'maize_yield_aug_sept_2018'
})

C:\Users\WeilunShi\AppData\Local\Temp\ipykernel_66708\1727251507.py:11: UnicodeWarning: 
One or more strings in the dta file could not be decoded using utf-8, and
so the fallback encoding of latin-1 is being used.  This can happen when a file
has been incorrectly encoded by Stata or some other software. You should verify
the string values returned are correct.
  globals()['df{}'.format(i+1)] = pd.read_stata(all_filenames[i])
C:\Users\WeilunShi\AppData\Local\Temp\ipykernel_66708\1727251507.py:11: UnicodeWarning: 
One or more strings in the dta file could not be decoded using utf-8, and
so the fallback encoding of latin-1 is being used.  This can happen when a file
has been incorrectly encoded by Stata or some other software. You should verify
the string values returned are correct.
  globals()['df{}'.format(i+1)] = pd.read_stata(all_filenames[i])


In [88]:
# describe df1 in detail
df1.describe().transpose()

count          mean           std   
HH_ID                             920.0  1.206077e+07  1.741254e+08  \
HOW_MANY_YEARS_HAVE_YOU_BEEN_A_M  920.0  4.260000e+00  3.608111e+00   
date                              920.0  1.549580e+07  9.490492e+06   
time                              920.0  1.276499e+05  3.946765e+04   
END_TIME                          918.0  1.322231e+05  4.066617e+04   
LI_LATITUDE                       918.0  8.287413e-01  1.063295e-01   
LI_LONGITUDE                      918.0  3.309696e+01  1.549812e+00   
ALTITUDE                          918.0  1.069981e+03  6.860081e+01   
household_size                    920.0  6.923913e+00  2.618814e+00   
category                          920.0  2.000000e+00  0.000000e+00   

                                        min           25%           50%   
HH_ID                              110104.0  3.200232e+05  5.603805e+05  \
HOW_MANY_YEARS_HAVE_YOU_BEEN_A_M        0.0  2.000000e+00  3.000000e+00   
date                              1102019.0  7.102019e+06  1.909202e+07   
time                                  631.0  1.107395e+05  1.309855e+05   
END_TIME                              107.0  1.150220e+05  1.352665e+05   
LI_LATITUDE                             0.0  7.399983e-01  8.268703e-01   
LI_LONGITUDE                            0.0  3.308850e+01  3.319402e+01   
ALTITUDE                                0.0  1.058000e+03  1.073000e+03   
household_size                          1.0  5.000000e+00  7.000000e+00   
category                                2.0  2.000000e+00  2.000000e+00   

                                           75%           max  
HH_ID                             8.000032e+05  3.500270e+09  
HOW_MANY_YEARS_HAVE_YOU_BEEN_A_M  5.000000e+00  3.800000e+01  
date                              2.409202e+07  3.009202e+07  
time                              1.526555e+05  2.357120e+05  
END_TIME                          1.606768e+05  2.356170e+05  
LI_LATITUDE                       9.253565e-01  1.069759e+00  
LI_LONGITUDE                      3.324012e+01  3.396016e+01  
ALTITUDE                          1.090000e+03  1.312000e+03  
household_size                    8.000000e+00  1.600000e+01  
category                          2.000000e+00  2.000000e+00

Nothing worth mentioning here.
0 in latitude, longitude and altitude could beconsidered as missing values.

In [4]:
# Is HH_ID unique?
df1['HH_ID'].nunique()
# There is no variation in 'category', so we drop it
df1 = df1.drop(columns=['category'])

In [87]:
df1.describe(include='object').transpose()

count unique            top freq
parish              920    166        Luzinga   34
village             920    406         bugobi   15
county              920     51         Buzaya  177
name                920    913  Nangobi Sarah    3
name_head           920    913  Nangobi Sarah    2
cell_head           920    843      999999999   57
cell_spouse         920    495      999999999  376
NAME_OF_THE_GROUP   920    451   Balitwegomba   19

999999999 should be missing values

In [89]:
df2.describe().transpose()

count          mean           std       min       25%       50%   
HH_ID    6370.0  1.239892e+07  1.761276e+08  110104.0  301238.0  560364.0  \
ID_CODE  6370.0  4.486342e+00  2.736026e+00       1.0       2.0       4.0   
age      6367.0  2.047178e+01  1.760370e+01       0.0       8.0      14.0   

              75%           max  
HH_ID    790182.0  3.500270e+09  
ID_CODE       6.0  1.600000e+01  
age          28.0  1.000000e+02

Age in df2 is ***unusually small***, others are normal

In [90]:
df2.describe(include='all').transpose()

count unique                       top  freq   
HH_ID                          6370.0    NaN                       NaN   NaN  \
ID_CODE                        6370.0    NaN                       NaN   NaN   
name                             6370   6217             Nabirye viola     5   
relationship                     6367      9              son/daughter  3583   
relationship_other               6370     17                            5475   
marital_status                   6367      5             never married  4568   
sex                              6369      3                      male  3184   
attend_school                    6369      2                       yes  3465   
education                        6369     23  never in school/ilitrate  1037   
usual_residence                  6369      4                       yes  6258   
guest                             111      2                        no   103   
stay_last_night                  6367      2                       yes  5724   
age                            6367.0    NaN                       NaN   NaN   
occupation                       6369      9      school/college child  3354   
occupation_other                 6370     51                            5431   
contribution_farming_activity    6367      3                 part time  3077   
add_member                       6367      2                       YES  5458   

                                          mean               std       min   
HH_ID                          12398922.907064  176127602.181136  110104.0  \
ID_CODE                               4.486342          2.736026       1.0   
name                                       NaN               NaN       NaN   
relationship                               NaN               NaN       NaN   
relationship_other                         NaN               NaN       NaN   
marital_status                             NaN               NaN       NaN   
sex                                        NaN               NaN       NaN   
attend_school                              NaN               NaN       NaN   
education                                  NaN               NaN       NaN   
usual_residence                            NaN               NaN       NaN   
guest                                      NaN               NaN       NaN   
stay_last_night                            NaN               NaN       NaN   
age                                  20.471776         17.603697       0.0   
occupation                                 NaN               NaN       NaN   
occupation_other                           NaN               NaN       NaN   
contribution_farming_activity              NaN               NaN       NaN   
add_member                                 NaN               NaN       NaN   

                                    25%       50%       75%           max  
HH_ID                          301238.0  560364.0  790182.0  3500269999.0  
ID_CODE                             2.0       4.0       6.0          16.0  
name                                NaN       NaN       NaN           NaN  
relationship                        NaN       NaN       NaN           NaN  
relationship_other                  NaN       NaN       NaN           NaN  
marital_status                      NaN       NaN       NaN           NaN  
sex                                 NaN       NaN       NaN           NaN  
attend_school                       NaN       NaN       NaN           NaN  
education                           NaN       NaN       NaN           NaN  
usual_residence                     NaN       NaN       NaN           NaN  
guest                               NaN       NaN       NaN           NaN  
stay_last_night                     NaN       NaN       NaN           NaN  
age                                 8.0      14.0      28.0         100.0  
occupation                          NaN       NaN       NaN           NaN  
occupation_other                    NaN       

In [8]:
# see relationship_other, and see unique values
df2['relationship_other'].unique()

array([' ', 'brother', 'parent', 'niece', 'sister', '3', '',
       'brother in law', 'Brother', 'Second wife',
       'child she looks after', 'Sister', 'Nephew', 'Neice', 'neice',
       'Brother inlaw', 'nephew'], dtype=object)

In [9]:
# se relationship, see unique values
df2['relationship'].unique()

['grandson/granddaughter', 'spouse', 'son/daughter', 'household head', 'parent', 'son/daughter-in-law', 'other relative', 'hired worker', 'other specify', NaN]
Categories (9, object): ['household head' < 'spouse' < 'son/daughter' < 'parent' ... 'grandson/granddaughter' < 'other relative' < 'hired worker' < 'other specify']

brother, sister should be listed in the previous question, as well as niece and nephew

In [10]:
# see unique occupation_other
df2['occupation_other'].unique()

array([' ', 'no occupation', 'no longer working,', 'Tailor', 'no work',
       'too old to work', 'bedridden', 'sick', 'not working',
       'works in salon', 'volunteer', 'Businessman', 'eldery', 'old age',
       'businessman', 'nothing', 'training  tailoring',
       'learning engineering', 'incapacitated', 'disabled', '4', '',
       'blind', 'fishing', 'building constructio', 'bodaboda but not his',
       'retired teacher', 'vocational studies', 'wage earner(driver)',
       'studying nursery tea', 'politician', 'none', 'pastor', '6',
       'physically handcappe', 'lumbering', 'charcoal burning', 'trader',
       'sickly', 'Sickly', 'at home sick', 'produce dealer', 'is sick',
       'House work', 'mechanic', 'taxi driver', 'attends to shop',
       'elderly', 'retired', 'home care work', 'taxi conductor'],
      dtype=object)

In [11]:
df2['occupation'].unique()

['school/college child', 'salaried employment', 'farming (crop+ livestock)', 'non-school child', 'other primary occupation, specify', 'casual labouer off-farm', 'self-employed off-farm', 'casual labourer on-farm', 0.0, NaN]
Categories (9, object): [0.0 < 'farming (crop+ livestock)' < 'salaried employment' < 'self-employed off-farm' ... 'casual labouer off-farm' < 'school/college child' < 'non-school child' < 'other primary occupation, specify']

Any sorts of no working or unemployment should be listed independently in the previous question, also retired/having no work ability should be as well listed

In [91]:
df3.describe().transpose()

count          mean           std   
HH_ID                                920.0  1.206077e+07  1.741254e+08  \
TOTAL_CULTIVATED_LAND_LAND           919.0  8.038659e+00  1.359940e+02   
total_land_cultivated_jul_aug_2019   919.0  3.808887e+00  3.158490e+00   
total_land_cultivated_feb_mar_2019   919.0  3.751728e+00  3.069835e+00   
total_land_cultivated_aug_sept_2018  919.0  3.694615e+00  3.083186e+00   
total_land_damaged_jul_aug_2019      919.0  6.747552e-01  1.779066e+00   
total_land_damaged_feb_mar_2019      919.0  6.350925e-01  1.754106e+00   
total_land_damaged_aug_sept_2018     919.0  5.776931e-01  1.721433e+00   
striga_seriousness                   689.0  8.959361e+00  1.227997e+01   
maize_area_jul_aug_2019              919.0  1.181352e+00  1.076899e+00   
maize_area_feb_mar_2019              919.0  1.318421e+00  3.307413e+00   
maize_area_aug_sept_2018             919.0  1.152625e+00  1.167565e+00   
maize_yield_jul_aug_2019             919.0  1.384113e+00  6.496254e-01   
maize_yield_feb_mar_2019             919.0  1.282916e+00  6.847560e-01   
maize_yield_aug_sept_2018            919.0  1.181719e+00  6.162760e-01   

                                          min        25%       50%        75%   
HH_ID                                110104.0  320023.25  560380.5  800003.25  \
TOTAL_CULTIVATED_LAND_LAND                0.0       1.50       3.0       4.50   
total_land_cultivated_jul_aug_2019        0.0       2.00       3.0       5.00   
total_land_cultivated_feb_mar_2019        0.0       2.00       3.0       5.00   
total_land_cultivated_aug_sept_2018       0.0       2.00       3.0       4.50   
total_land_damaged_jul_aug_2019           0.0       0.00       0.0       1.00   
total_land_damaged_feb_mar_2019           0.0       0.00       0.0       1.00   
total_land_damaged_aug_sept_2018          0.0       0.00       0.0       1.00   
striga_seriousness                        0.0       5.00       5.0       8.00   
maize_area_jul_aug_2019                   0.0       0.50       1.0       1.50   
maize_area_feb_mar_2019                   0.0       0.50       1.0       1.50   
maize_area_aug_sept_2018                  0.0       0.50       1.0       1.50   
maize_yield_jul_aug_2019                  0.0       1.00       1.0       2.00   
maize_yield_feb_mar_2019                  0.0       1.00       1.0       2.00   
maize_yield_aug_sept_2018                 0.0       1.00       1.0       1.00   

                                              max  
HH_ID                                3.500270e+09  
TOTAL_CULTIVATED_LAND_LAND           4.125000e+03  
total_land_cultivated_jul_aug_2019   4.200000e+01  
total_land_cultivated_feb_mar_2019   4.200000e+01  
total_land_cultivated_aug_sept_2018  4.100000e+01  
total_land_damaged_jul_aug_2019      4.100000e+01  
total_land_damaged_feb_mar_2019      4.100000e+01  
total_land_damaged_aug_sept_2018     4.000000e+01  
striga_seriousness                   9.000000e+01  
maize_area_jul_aug_2019              2.000000e+01  
maize_area_feb_mar_2019              9.500000e+01  
maize_area_aug_sept_2018             2.000000e+01  
maize_yield_jul_aug_2019             4.000000e+00  
maize_yield_feb_mar_2019             1.200000e+01  
maize_yield_aug_sept_2018            4.000000e+00

In [92]:
df3.describe(include='all').transpose()

count unique             top freq   
HH_ID                                920.0    NaN             NaN  NaN  \
TOTAL_CULTIVATED_LAND_LAND           919.0    NaN             NaN  NaN   
HH_crop_1                              919     16           Maize  630   
HH_crop_2                              919     23           Beans  265   
HH_crop_3                              918     22           Beans  179   
HH_crop_4                              913     25           Beans  164   
HH_crop_5                              903     25  Sweet potatoes  138   
HH_crop_other                          920     15                  894   
total_land_cultivated_jul_aug_2019   919.0    NaN             NaN  NaN   
total_land_cultivated_feb_mar_2019   919.0    NaN             NaN  NaN   
total_land_cultivated_aug_sept_2018  919.0    NaN             NaN  NaN   
total_land_damaged_jul_aug_2019      919.0    NaN             NaN  NaN   
total_land_damaged_feb_mar_2019      919.0    NaN             NaN  NaN   
total_land_damaged_aug_sept_2018     919.0    NaN             NaN  NaN   
FAW_awareness                          919      2             Yes  895   
Recognize_FAW                          895      2             Yes  689   
FAW_major_constraint                   895      2             Yes  827   
stem_borer_awareness                   919      2             Yes  896   
Recognize_stem_borer                   896      2             Yes  708   
stem_borer_major_constraint            896      2             Yes  498   
striga_constraint                      919      2             Yes  689   
striga_seriousness                   689.0    NaN             NaN  NaN   
maize_area_jul_aug_2019              919.0    NaN             NaN  NaN   
maize_area_feb_mar_2019              919.0    NaN             NaN  NaN   
maize_area_aug_sept_2018             919.0    NaN             NaN  NaN   
maize_yield_jul_aug_2019             919.0    NaN             NaN  NaN   
maize_yield_feb_mar_2019             919.0    NaN             NaN  NaN   
maize_yield_aug_sept_2018            919.0    NaN             NaN  NaN   

                                                mean               std   
HH_ID                                12060774.104348  174125413.075476  \
TOTAL_CULTIVATED_LAND_LAND                  8.038659        135.993959   
HH_crop_1                                        NaN               NaN   
HH_crop_2                                        NaN               NaN   
HH_crop_3                                        NaN               NaN   
HH_crop_4                                        NaN               NaN   
HH_crop_5                                        NaN               NaN   
HH_crop_other                                    NaN               NaN   
total_land_cultivated_jul_aug_2019          3.808887           3.15849   
total_land_cultivated_feb_mar_2019          3.751728          3.069835   
total_land_cultivated_aug_sept_2018         3.694615          3.083186   
total_land_damaged_jul_aug_2019             0.674755          1.779066   
total_land_damaged_feb_mar_2019             0.635092          1.754106   
total_land_damaged_aug_sept_2018            0.577693          1.721433   
FAW_awareness                                    NaN               NaN   
Recognize_FAW                                    NaN               NaN   
FAW_major_constraint                             NaN               NaN   
stem_borer_awareness                             NaN               NaN   
Recognize_stem_borer                             NaN               NaN   
stem_borer_major_constraint                      NaN               NaN   
striga_constraint                                NaN               NaN   
striga_seriousness                          8.959361         12.279974   
maize_area_jul_aug_2019                     1.181352          1.076899   
maize_area_feb_mar_2019                     1.318421          3.307413   
maize_area_aug_sept_2018                

In [14]:
# see percentage of HH_crop_1
df3['HH_crop_1'].value_counts(normalize=True)

HH_crop_1
Maize                  0.685528
Coffee                 0.108814
Sugar cane             0.064200
Sweet potatoes         0.034820
Beans                  0.030468
Banana                 0.023939
Rice                   0.019587
Groundnuts             0.006529
Tomatoes               0.006529
Cocoa                  0.004353
Millet                 0.004353
Soybeans               0.004353
Cassava                0.003264
Eggplants              0.001088
Yams                   0.001088
Other crop, specify    0.001088
Name: proportion, dtype: float64

In [93]:
df4.describe().transpose()

count          mean           std        min       25%       50%   
HH_ID      1721.0  8.355908e+06  1.359680e+08  110104.00  301238.0  560381.0  \
PLOT_ID    1721.0  1.748983e+00  1.026838e+00       1.00       1.0       1.0   
PLOT_AREA  1721.0  1.815294e+00  2.251900e+00       0.01       0.5       1.0   

                75%           max  
HH_ID      790185.0  3.500270e+09  
PLOT_ID         2.0  6.000000e+00  
PLOT_AREA       2.0  4.000000e+01

In [16]:
# describe and show all columns,transposed
pd.set_option('display.max_rows', None)
df8.describe(include='all').transpose()

count unique   
id                                10599.0    NaN  \
num                               10599.0    NaN   
HH_ID                             10599.0    NaN   
SERIAL_NO                         10599.0    NaN   
SEASON_NAME                         10599      3   
PLOTID                            10599.0    NaN   
PLOT_LOCATION_NAME                  10599    644   
PLOTAREA                          10599.0    NaN   
DID_YOU_HAVE_ANOTHER_MAIZE_PLOT     10599      2   
SERIAL_NQ                         10599.0    NaN   
SEASON                              10599      3   
SUBPLOTID                         10599.0    NaN   
Plot_Name                           10599   1181   
SUBPLOTarea                       10599.0    NaN   
SUB_PLOT_TENURE                     10599      5   
OTHER_SUB_PLOT_TENURE_SPECIFY       10599      5   
M3A6B                               10599      7   
PLOT_DISTANCE_TO_RESIDENCE_WALK   10599.0    NaN   
INTERCROPPING_ON_THIS_PLOT          10599      2   
NUMBER_OF_CROPS_GROWN                5664      3   
FIRST_MAIN_CROP_GROWN                5664     17   
M3A10_OTHER_SPECIFY1                10599      5   
PERCENT_OF_AREA_FOR_FIRST_MAIN_C   5664.0    NaN   
SECOND_MAIN_CROP_GROWO               1698     17   
M3A12_OTHER_SPECIFY                 10599      5   
PERCENT_OF_AREA_FOR_SECOND_MAIN    1899.0    NaN   
PLANTING_METHOD                     10599      2   
HOW_FERTILE_IS_THE_SOIL_OF_THIS     10599      3   
WHAT_IS_THE_SOIL_SLOPE_OF_THIT      10599      3   
WHAT_IS_THE_SOIL_DEPTH_OF_THIS      10599      3   
MAIN_CROP_GROWN_DURING_PREVIOUS     10599     30   
OTHER_MAIN_CROP_SPECIFY             10599      2   
M3B7                                10599      2   
OTHER_SOIL_WATER_CONSERVATION       10599     14   
OTHER_SOIL_WATER_CONSERVATION1      10599      7   
OTHER_SOIL_WATER_CONSERVATION2      10599      3   
OTHE_SOIL_AND_WATER_CONSERVATION    10599      6   
SUB_PLOT_IRRIGATEE                  10599      2   
HAS_THIS_PLOT_RECEIVED_INSECTICI    10599      2   
M3C5B                                2370      6   
TOTAL_QUANTITY_USED_KG_OR_LITTER   2370.0    NaN   
TOTAL_COST_OF_INSECTICIDE_USED_E   2370.0    NaN   
HAS_THIS_PLOT_RECEIVED_HERBICIDG    10599      2   
C6B_TYPE_OF_HERBICIDES              10599     42   
TOTAL_QUANTITY_OF_HERBICIDES_USF    660.0    NaN   
TOTAL_COST_OF_HERBICIDES_USED_EU    660.0    NaN   
HAS_THIS_PLOT_RECEIVED_URA_1_YG     10599      2   
TOTAL_QUANTITY_OF_URA_USED_KG_P    1377.0    NaN   
TOTAL_COST_OF_URA_USED_ETC         1377.0    NaN   
HAS_THIS_PLOT_RECEIVED_DAP_1_YF     10599      2   
TOTAL_QUANTITY_OF_DAP_USED_KG_O    1836.0    NaN   
TOTAL_COST_OF_DAP_USED_ETB         1836.0    NaN   
HAS_THIS_PLOT_RECEIVED_MANURE_A     10599      2   
QUANTITY_OF_NON_BOUGHT_OWN_SAVF   10599.0    NaN   
QUANTITY_OF_BOUGHT_MAIZE_SEED_J   10599.0    NaN   
TOTAL_COST_OF_BOUGHT_MAIZE_SEEE   10599.0    NaN   
TYPE_OF_MAIZE_SEED_USEE             10599      3   
M3D5A                               10599      3   
M3D5B                             10599.0    NaN   
TOTAL_COST_OF_HIRED_OXEN_UGX       1959.0    NaN   
TOTAL_COST_OF_HIRED_TRACTOR_UGX      81.0    NaN   
M3D5C                             10599.0    NaN   
M3D5D                             10599.0    NaN   
M3D5E                             10599.0    NaN   
M3D5F                             10599.0    NaN   
M3D6A                             10599.0    NaN   
M3D6B                             10599.0    NaN   
M3D6C                             10599.0    NaN   
M3D6D                             10599.0    NaN   
M3D6E                             10599.0    NaN   
M3D7A                             10416.0    NaN   
M3D7B                             10416.0    NaN   
M3D7C                             10416.0    NaN   
M3D7D                             10416.0    NaN   
M3D8A                             10416.0    NaN   
M3D8B                             10416.0    NaN   
M3D8C                           

Insectide has 6 values, while herbicide has 42 values.

In [17]:
# see percentage of TYPE_OF_MAIZE_SEED_USEE
df8['TYPE_OF_MAIZE_SEED_USEE'].value_counts(normalize=True)

TYPE_OF_MAIZE_SEED_USEE
Local       0.639400
Improved    0.341636
Both        0.018964
Name: proportion, dtype: float64

M3D4, almost all maize seeds come from local, what does improved mean?

In [18]:
# see percentage of M3E4A:Did crop suffer from FAW
df8['M3E4A'].value_counts(normalize=True)

M3E4A
Yes    0.739315
No     0.260685
Name: proportion, dtype: float64

In [19]:
#see percentage of M3E5A:strategy used to control FAW
df8['M3E5A'].value_counts(normalize=True)

M3E5A
 mechanically remove the insect pest    0.327117
 Pesticides application                 0.233868
Uprooting and burning                   0.197687
 Overweeding/ Frequent weeding          0.143230
Ash/pepper                              0.057068
intercropping                           0.015293
Early planting                          0.014547
crop rotation                           0.004849
manure application                      0.004849
DDT                                     0.001119
Malathion                               0.000373
Name: proportion, dtype: float64

In [20]:
# see percentage of M3E6:Did crop suffer from stem borer
df8['M3E6'].value_counts(normalize=True)

M3E6
No     0.604868
Yes    0.395132
Name: proportion, dtype: float64

Not so many

In [21]:
# see percentage of M3E8A:strategy used to control stem borer
df8['M3E8A'].value_counts(normalize=True)

M3E8A
 mechanically remove the insect pest    0.495307
 Pesticides application                 0.156679
Uprooting and burning                   0.144404
 Overweeding/ Frequent weeding          0.109747
Early planting                          0.038989
Ash/pepper                              0.035379
intercropping                           0.010108
manure application                      0.005776
crop rotation                           0.003610
Name: proportion, dtype: float64

In [22]:
# see percentage of M3E9:Did crop suffer from striga
df8['M3E9'].value_counts(normalize=True)

M3E9
Yes    0.556185
No     0.443815
Name: proportion, dtype: float64

In [23]:
# see percentage of M3E13A: other major threat
df8['M3E13A'].value_counts(normalize=True)

M3E13A
     0.486555
G    0.278800
D    0.129069
L    0.031701
K    0.015851
A    0.014435
B    0.014152
F    0.011605
E    0.006227
I    0.003680
J    0.003397
     0.002547
C    0.001415
H    0.000566
Name: proportion, dtype: float64

CODE 2
1. No Stress
2. Soil loss 
3. Striga (show pic)
4. other Insects/pests
5.Disease	5.WaterLogging
6.Drought
7.Frost
8. Flood	9.Hailstorm
10.Animal trampling
11.Other, specify…………….


M3E14	5745.0	NaN	NaN	NaN	583.85483	1425.659124	0.0	50.0	250.0	550.0	9999.0
M3E5A	2681	11	mechanically remove the insect pest	877	NaN	NaN	NaN	NaN	NaN	NaN	NaN
M3E8A	1385	9	mechanically remove the insect pest	686	NaN	NaN	NaN	NaN	NaN	NaN	NaN
M3E11A	2565	9	Overweeding/ Frequent weeding	1539	NaN	NaN	NaN	NaN	NaN	NaN	NaN
Season	10599	3	current season (Aug/Sept 2019 planting)	3861	NaN	NaN	NaN	NaN	NaN	NaN	NaN

Not very meaningful

In [25]:
df9.describe(include='all').transpose()

count unique   
HH_ID               2856.0    NaN  \
M3F1                  1941      3   
M3F2                  1941      4   
M3F2_OTHER_SPECIFY    2856     18   
M3F3                1941.0   19.0   
M3F3A               1941.0   11.0   
M3F3B               1808.0   15.0   
M3F5                1786.0    NaN   
M3F6                1786.0    NaN   
add_other_season        13      2   

                                                            top   freq   
HH_ID                                                       NaN    NaN  \
M3F1                Long rain season (Feb/Marchl 2019 planting)    911   
M3F2                                                  Farm gate   1301   
M3F2_OTHER_SPECIFY                                                2626   
M3F3                                                       14.0  609.0   
M3F3A                                                      14.0  611.0   
M3F3B                                                      14.0  613.0   
M3F5                                                        NaN    NaN   
M3F6                                                        NaN    NaN   
add_other_season                                             No      7   

                              mean              std       min       25%   
HH_ID               7869515.504202  139672260.26105  110104.0  330002.0  \
M3F1                           NaN              NaN       NaN       NaN   
M3F2                           NaN              NaN       NaN       NaN   
M3F2_OTHER_SPECIFY             NaN              NaN       NaN       NaN   
M3F3                           NaN              NaN       NaN       NaN   
M3F3A                          NaN              NaN       NaN       NaN   
M3F3B                          NaN              NaN       NaN       NaN   
M3F5                     434.56047      3004.233102       0.0       0.0   
M3F6                  51768.095745    216627.338648       0.0       0.0   
add_other_season               NaN              NaN       NaN       NaN   

                         50%       75%           max  
HH_ID               580420.0  790181.0  3500269999.0  
M3F1                     NaN       NaN           NaN  
M3F2                     NaN       NaN           NaN  
M3F2_OTHER_SPECIFY       NaN       NaN           NaN  
M3F3                     NaN       NaN           NaN  
M3F3A                    NaN       NaN           NaN  
M3F3B                    NaN       NaN           NaN  
M3F5                   100.0     500.0      120000.0  
M3F6                   500.0     800.0     5000000.0  
add_other_season         NaN       NaN           NaN

In [26]:
# see percentage of M3F2: market place
df9['M3F2'].value_counts(normalize=True)

M3F2
Farm gate               0.670273
Other (specify)         0.267903
Village market          0.055126
Main/district market    0.006698
Name: proportion, dtype: float64

In [27]:
# see percentage of M3F2_OTHER_SPECIFY: market place
df9['M3F2_OTHER_SPECIFY'].value_counts(normalize=True)

M3F2_OTHER_SPECIFY
                     0.919468
                     0.067227
not sold             0.003852
did not sell         0.001401
no maize sales       0.001401
did not sale         0.001050
dont sell maize      0.000700
not harvested yet    0.000700
dint sell            0.000700
never sold           0.000700
dont sell maize,     0.000350
dnt know sell        0.000350
have nt harvested    0.000350
not yet harvested    0.000350
no sales made        0.000350
didnot sell          0.000350
didinot sell         0.000350
0                    0.000350
Name: proportion, dtype: float64

so did not sell should be an independent choice

In [29]:
df10.describe(include='all').transpose()

count unique             top freq   
HH_ID                             920.0    NaN             NaN  NaN  \
A1                                  920     11         Wankole  156   
A2                                  920    166         Luzinga   34   
A3                                  920    406          bugobi   15   
A4                                  920     51          Buzaya  177   
A6                                  920      2          Female  530   
A8                                  920      2            Male  787   
A9                                  772      4  Household head  472   
NAME_OF_THE_GROUP                   920    451    Balitwegomba   19   
POSITION_IN_THE_GROUP               920      2          Member  667   
HOW_MANY_YEARS_HAVE_YOU_BEEN_A_M  920.0    NaN             NaN  NaN   
M31E_I                              919      2             Yes  895   
M31E_II                             895      2             Yes  689   
M31E_III                            895      2             Yes  827   
M31F_I                              919      2             Yes  896   
M31F_II                             896      2             Yes  708   
M31F_III                            896      2             Yes  498   
M31G                                919      2             Yes  689   
M31H                              689.0    NaN             NaN  NaN   
SUPPORT_FAW                         919      2              No  693   
SUPPORT_STIGA                       919      2              No  687   
SUPPORT_STEMBORE                    919      2              No  781   

                                             mean               std       min   
HH_ID                             12060774.104348  174125413.075476  110104.0  \
A1                                            NaN               NaN       NaN   
A2                                            NaN               NaN       NaN   
A3                                            NaN               NaN       NaN   
A4                                            NaN               NaN       NaN   
A6                                            NaN               NaN       NaN   
A8                                            NaN               NaN       NaN   
A9                                            NaN               NaN       NaN   
NAME_OF_THE_GROUP                             NaN               NaN       NaN   
POSITION_IN_THE_GROUP                         NaN               NaN       NaN   
HOW_MANY_YEARS_HAVE_YOU_BEEN_A_M             4.26          3.608111       0.0   
M31E_I                                        NaN               NaN       NaN   
M31E_II                                       NaN               NaN       NaN   
M31E_III                                      NaN               NaN       NaN   
M31F_I                                        NaN               NaN       NaN   
M31F_II                                       NaN               NaN       NaN   
M31F_III                                      NaN               NaN       NaN   
M31G                                          NaN               NaN       NaN   
M31H                                     8.959361         12.279974       0.0   
SUPPORT_FAW                                   NaN               NaN       NaN   
SUPPORT_STIGA                                 NaN               NaN       NaN   
SUPPORT_STEMBORE                              NaN               NaN       NaN   

                                        25%       50%        75%           max  
HH_ID                             320023.25  560380.5  800003.25  3500269999.0  
A1                                      NaN       NaN        NaN           NaN  
A2                                      NaN       NaN        NaN           NaN  
A3                                      NaN       NaN        NaN           NaN  
A4                                      NaN       NaN        NaN           NaN  
A6                                      NaN       

In [30]:
# see percentage of M31E_1: FAW awareness
df10['M31E_I'].value_counts(normalize=True)

M31E_I
Yes    0.973885
No     0.026115
Name: proportion, dtype: float64

In [31]:
# see percentage of M31E_2: Recognize FAW
df10['M31E_II'].value_counts(normalize=True)

M31E_II
Yes    0.769832
No     0.230168
Name: proportion, dtype: float64

In [32]:
# see percentage of M31E_3: FAW major constraint
df10['M31E_III'].value_counts(normalize=True)

M31E_III
Yes    0.924022
No     0.075978
Name: proportion, dtype: float64

In [33]:
# see percentage of M31F_1: stem borer awareness
df10['M31F_I'].value_counts(normalize=True)

M31F_I
Yes    0.974973
No     0.025027
Name: proportion, dtype: float64

In [34]:
# see percentage of M31F_2: Recognize stem borer
df10['M31F_II'].value_counts(normalize=True)

M31F_II
Yes    0.790179
No     0.209821
Name: proportion, dtype: float64

In [35]:
# see percentage of M31F_3: stem borer major constraint
df10['M31F_III'].value_counts(normalize=True)

M31F_III
Yes    0.555804
No     0.444196
Name: proportion, dtype: float64

In [36]:
# see percentage of M31G: striga constraint
df10['M31G'].value_counts(normalize=True)

M31G
Yes    0.749728
No     0.250272
Name: proportion, dtype: float64

In [37]:
# see percentage of M31H: striga seriousness
df10['M31H'].value_counts(normalize=True)


M31H
5.0     0.274311
8.0     0.095791
3.0     0.094340
6.0     0.092888
10.0    0.089985
7.0     0.082729
4.0     0.049347
9.0     0.049347
2.0     0.047896
1.0     0.027576
50.0    0.020319
30.0    0.015965
20.0    0.010160
0.0     0.010160
25.0    0.008708
60.0    0.005806
40.0    0.005806
80.0    0.005806
70.0    0.004354
35.0    0.002903
65.0    0.001451
85.0    0.001451
90.0    0.001451
15.0    0.001451
Name: proportion, dtype: float64

In [38]:
# see percentage of SUPPORT_FAW: support FAW
df10['SUPPORT_FAW'].value_counts(normalize=True)

SUPPORT_FAW
No     0.754081
Yes    0.245919
Name: proportion, dtype: float64

In [40]:
# see percentage of SUPPORT_STEM_BORER: support stem borer
df10['SUPPORT_STEMBORE'].value_counts(normalize=True)

SUPPORT_STEMBORE
No     0.849837
Yes    0.150163
Name: proportion, dtype: float64

In [42]:
# see percentage of SUPPORT_STIGA
df10['SUPPORT_STIGA'].value_counts(normalize=True)

SUPPORT_STIGA
No     0.747552
Yes    0.252448
Name: proportion, dtype: float64

In [44]:
df11.describe(include='all').transpose()

count unique                            top freq   
HH_ID                     325.0    NaN                            NaN  NaN  \
SUPPORT_RECEIVED_FOR_FAW    325      5  Training on proper management  131   
FAW_SUPPORT_OTHER           325      6                                 317   
INSTIUTIONS_SUPPORT_FAW     325      4                  One acre fund  162   
INSTITUTION_FAW_OTHER       325     40                                 260   
ADD_OTHER_SUPPORT           325      2                             No  220   

                                     mean               std       min   
HH_ID                     10039279.273846  110088375.120839  110106.0  \
SUPPORT_RECEIVED_FOR_FAW              NaN               NaN       NaN   
FAW_SUPPORT_OTHER                     NaN               NaN       NaN   
INSTIUTIONS_SUPPORT_FAW               NaN               NaN       NaN   
INSTITUTION_FAW_OTHER                 NaN               NaN       NaN   
ADD_OTHER_SUPPORT                     NaN               NaN       NaN   

                               25%       50%       75%           max  
HH_ID                     270001.0  610110.0  750209.0  1400019999.0  
SUPPORT_RECEIVED_FOR_FAW       NaN       NaN       NaN           NaN  
FAW_SUPPORT_OTHER              NaN       NaN       NaN           NaN  
INSTIUTIONS_SUPPORT_FAW        NaN       NaN       NaN           NaN  
INSTITUTION_FAW_OTHER          NaN       NaN       NaN           NaN  
ADD_OTHER_SUPPORT              NaN       NaN       NaN           NaN

In [45]:
# see percentage of SUPPORT_RECEIVED_FOR_FAW: support received for FAW
df11['SUPPORT_RECEIVED_FOR_FAW'].value_counts(normalize=True)

SUPPORT_RECEIVED_FOR_FAW
Training on proper management    0.403077
Information and advisory         0.341538
Pesticides                       0.196923
Improved seed                    0.046154
Others (specify)                 0.012308
Name: proportion, dtype: float64

In [46]:
# see percentage of FAW_SUPPORT_OTHER: support received for FAW
df11['FAW_SUPPORT_OTHER'].value_counts(normalize=True)

FAW_SUPPORT_OTHER
                   0.975385
                   0.012308
managing faw       0.003077
Radio programme    0.003077
fellow farmers     0.003077
radio              0.003077
Name: proportion, dtype: float64

In [47]:
# see percentage of INSTIUTIONS_SUPPORT_FAW: support received for FAW
df11['INSTIUTIONS_SUPPORT_FAW'].value_counts(normalize=True)

INSTIUTIONS_SUPPORT_FAW
One acre fund             0.498462
NAADs/Local Gioverment    0.221538
Others (specify)          0.193846
Sasakwa global 2000       0.086154
Name: proportion, dtype: float64

In [48]:
# see percentage of INSTITUTION_FAW_OTHER: support received for FAW
df11['INSTITUTION_FAW_OTHER'].value_counts(normalize=True)

INSTITUTION_FAW_OTHER
                             0.800000
kadifa                       0.024615
fellow farmer                0.012308
khadifa                      0.012308
send a cow                   0.009231
field officer                0.009231
kalana organisation          0.006154
e&c organisation             0.006154
fellow farmer in group       0.006154
vedco                        0.006154
fellow farmers               0.006154
mass media                   0.006154
dont remember                0.006154
group from makerere          0.006154
                             0.006154
world food programme         0.003077
Tibukake farmers             0.003077
agricultural officer         0.003077
Kaida                        0.003077
radio station                0.003077
bakuseka majja farmer gro    0.003077
VEDCO                        0.003077
karana                       0.003077
abi                          0.003077
kaida                        0.003077
nsanja chemicals on radio   

In [50]:
df12.describe(include='all').transpose()

count unique                            top   
HH_ID                           180.0    NaN                            NaN  \
SUPPORT_RECEIVED_FOR_STEMBORER    180      5  Training on proper management   
STEMBORER_SUPPORT_OTHER           180      5                                  
INSTIUTIONS_SUPPORT_STEMBORER     180      4                  One acre fund   
INSTITUTION_STEMBORER_OTHER       180     20                                  
ADD_OTHER_SUPPORT2                180      2                             No   

                               freq            mean              std   
HH_ID                           NaN  2128376.661111  14660308.299576  \
SUPPORT_RECEIVED_FOR_STEMBORER   94             NaN              NaN   
STEMBORER_SUPPORT_OTHER         173             NaN              NaN   
INSTIUTIONS_SUPPORT_STEMBORER    89             NaN              NaN   
INSTITUTION_STEMBORER_OTHER     149             NaN              NaN   
ADD_OTHER_SUPPORT2              135             NaN              NaN   

                                     min        25%       50%       75%   
HH_ID                           110106.0  250005.75  640106.5  770146.0  \
SUPPORT_RECEIVED_FOR_STEMBORER       NaN        NaN       NaN       NaN   
STEMBORER_SUPPORT_OTHER              NaN        NaN       NaN       NaN   
INSTIUTIONS_SUPPORT_STEMBORER        NaN        NaN       NaN       NaN   
INSTITUTION_STEMBORER_OTHER          NaN        NaN       NaN       NaN   
ADD_OTHER_SUPPORT2                   NaN        NaN       NaN       NaN   

                                        max  
HH_ID                           140009999.0  
SUPPORT_RECEIVED_FOR_STEMBORER          NaN  
STEMBORER_SUPPORT_OTHER                 NaN  
INSTIUTIONS_SUPPORT_STEMBORER           NaN  
INSTITUTION_STEMBORER_OTHER             NaN  
ADD_OTHER_SUPPORT2                      NaN

In [51]:
# see percentage of SUPPORT_RECEIVED_FOR_STEMBORER
df12['SUPPORT_RECEIVED_FOR_STEMBORER'].value_counts(normalize=True) 

SUPPORT_RECEIVED_FOR_STEMBORER
Training on proper management    0.522222
Information and advisory         0.361111
Pesticides                       0.055556
Improved seed                    0.033333
Others (specify)                 0.027778
Name: proportion, dtype: float64

In [55]:
# see percentage of STEM_BORER_SUPPORT_OTHER
df12['STEMBORER_SUPPORT_OTHER'].value_counts(normalize=True)

STEMBORER_SUPPORT_OTHER
                 0.961111
radio station    0.016667
                 0.011111
kadifa           0.005556
radio            0.005556
Name: proportion, dtype: float64

In [56]:
# see percentage of INSTIUTIONS_SUPPORT_STEMBORER
df12['INSTIUTIONS_SUPPORT_STEMBORER'].value_counts(normalize=True)

INSTIUTIONS_SUPPORT_STEMBORER
One acre fund             0.494444
NAADs/Local Gioverment    0.211111
Others (specify)          0.172222
Sasakwa global 2000       0.122222
Name: proportion, dtype: float64

In [57]:
# see percentage of INSTITUTION_STEMBORER_OTHER
df12['INSTITUTION_STEMBORER_OTHER'].value_counts(normalize=True)

INSTITUTION_STEMBORER_OTHER
                             0.827778
mass media                   0.022222
send a cow                   0.022222
Abi Trust                    0.011111
fellow farmer in group       0.011111
kalana organisation          0.011111
e&c organisation             0.011111
dont remember                0.011111
kadifa                       0.011111
KDFA                         0.005556
input dealers                0.005556
Tibukake farmers             0.005556
on radio, by nsanja chemi    0.005556
Aand E                       0.005556
Vedco                        0.005556
vedco                        0.005556
bakuseka majja farmer gro    0.005556
radio station                0.005556
kaida                        0.005556
fellow farmers               0.005556
Name: proportion, dtype: float64

In [58]:
df13.describe(include='all').transpose()

count unique                            top   
HH_ID                             301.0    NaN                            NaN  \
SUPPORT_RECEIVED_ON_STRIGA          301      5  Training on proper management   
STRIGA_SUPPORT_OTHER                301      9                                  
INSTIUTIONS_THAT_PROVIDED_THE_ST    301      5                  One acre fund   
INSTITUTION_STRIGA_OTHER            301     40                                  
ADD_OTHER_SUPPORT1                  301      2                             No   

                                 freq             mean               std   
HH_ID                             NaN  10761446.707641  114376232.993096  \
SUPPORT_RECEIVED_ON_STRIGA        156              NaN               NaN   
STRIGA_SUPPORT_OTHER              289              NaN               NaN   
INSTIUTIONS_THAT_PROVIDED_THE_ST  148              NaN               NaN   
INSTITUTION_STRIGA_OTHER          239              NaN               NaN   
ADD_OTHER_SUPPORT1                229              NaN               NaN   

                                       min       25%       50%       75%   
HH_ID                             110106.0  240005.0  520413.0  750163.0  \
SUPPORT_RECEIVED_ON_STRIGA             NaN       NaN       NaN       NaN   
STRIGA_SUPPORT_OTHER                   NaN       NaN       NaN       NaN   
INSTIUTIONS_THAT_PROVIDED_THE_ST       NaN       NaN       NaN       NaN   
INSTITUTION_STRIGA_OTHER               NaN       NaN       NaN       NaN   
ADD_OTHER_SUPPORT1                     NaN       NaN       NaN       NaN   

                                           max  
HH_ID                             1400019999.0  
SUPPORT_RECEIVED_ON_STRIGA                 NaN  
STRIGA_SUPPORT_OTHER                       NaN  
INSTIUTIONS_THAT_PROVIDED_THE_ST           NaN  
INSTITUTION_STRIGA_OTHER                   NaN  
ADD_OTHER_SUPPORT1                         NaN

In [60]:
# see percentage of SUPPORT_RECEIVED_ON_STRIGA
df13['SUPPORT_RECEIVED_ON_STRIGA'].value_counts(normalize=True)

SUPPORT_RECEIVED_ON_STRIGA
Training on proper management    0.518272
Information and advisory         0.418605
Others (specify)                 0.026578
Improved seed                    0.023256
Pesticides                       0.013289
Name: proportion, dtype: float64

In [61]:
# see percentage of STRIGA_SUPPORT_OTHER
df13['STRIGA_SUPPORT_OTHER'].value_counts(normalize=True)

STRIGA_SUPPORT_OTHER
                             0.960133
                             0.013289
radio station                0.006645
sprayer                      0.003322
karana                       0.003322
weeding and burning strig    0.003322
kadifa                       0.003322
fellow farmers               0.003322
radio                        0.003322
Name: proportion, dtype: float64

In [63]:
# see percentage of INSTIUTIONS_SUPPORT_STRIGA
df13['INSTIUTIONS_THAT_PROVIDED_THE_ST'].value_counts(normalize=True)

INSTIUTIONS_THAT_PROVIDED_THE_ST
One acre fund             0.491694
NAADs/Local Gioverment    0.225914
Others (specify)          0.199336
Sasakwa global 2000       0.079734
Food for the hungry       0.003322
Name: proportion, dtype: float64

In [64]:
# see percentage of INSTITUTION_STRIGA_OTHER	
df13['INSTITUTION_STRIGA_OTHER'].value_counts(normalize=True)

INSTITUTION_STRIGA_OTHER
                             0.794020
send a cow                   0.019934
khadifa                      0.019934
mass media                   0.009967
doesn't remember the orga    0.009967
lunch for learning           0.006645
dont remember                0.006645
e&c organisation             0.006645
fellow farmer in group       0.006645
field officer                0.006645
vedco                        0.006645
kadifa                       0.006645
                             0.006645
agro ways                    0.006645
kalana organisation          0.003322
karana                       0.003322
A znd E                      0.003322
kaidi                        0.003322
radio station                0.003322
kaida                        0.003322
don't know                   0.003322
bakuseka majja farmer gro    0.003322
Vedco                        0.003322
ORDS                         0.003322
Abi Trust                    0.003322
Kadifa                   

In [67]:
df14.describe(include='all').transpose()

count unique                        top   
HH_ID                             26651.0    NaN                        NaN  \
SERIAL_NUMBER1                    26651.0    NaN                        NaN   
INCOME_SOURCE                       26651     29  Transportation businesses   
DID_THE_HOUSEHOLD_ERN_INCOME_FRO    26651      2                         No   
OTHER_SPECIFY                       26651    114                              
HOW_MUCH_DID_THE_HOUSEHOLD_EARN_   3045.0    NaN                        NaN   
HOW_MUCH_DID_THE_HOUSEHOLDS_RECE   3045.0    NaN                        NaN   
TOTAL_INCOME                       3045.0    NaN                        NaN   

                                   freq             mean               std   
HH_ID                               NaN  12073027.243743  174128283.295801  \
SERIAL_NUMBER1                      NaN             15.0          8.366757   
INCOME_SOURCE                       919              NaN               NaN   
DID_THE_HOUSEHOLD_ERN_INCOME_FRO  23606              NaN               NaN   
OTHER_SPECIFY                     26161              NaN               NaN   
HOW_MUCH_DID_THE_HOUSEHOLD_EARN_    NaN    790655.573727    2193512.623388   
HOW_MUCH_DID_THE_HOUSEHOLDS_RECE    NaN     10632.950739      153170.53155   
TOTAL_INCOME                        NaN    801288.524466    2196647.875792   

                                       min       25%       50%       75%   
HH_ID                             110104.0  320018.0  560380.0  800003.0  \
SERIAL_NUMBER1                         1.0       8.0      15.0      22.0   
INCOME_SOURCE                          NaN       NaN       NaN       NaN   
DID_THE_HOUSEHOLD_ERN_INCOME_FRO       NaN       NaN       NaN       NaN   
OTHER_SPECIFY                          NaN       NaN       NaN       NaN   
HOW_MUCH_DID_THE_HOUSEHOLD_EARN_     -99.0  100000.0  300000.0  800000.0   
HOW_MUCH_DID_THE_HOUSEHOLDS_RECE       0.0       0.0       0.0       0.0   
TOTAL_INCOME                         -99.0  100000.0  300000.0  800000.0   

                                           max  
HH_ID                             3500269999.0  
SERIAL_NUMBER1                            29.0  
INCOME_SOURCE                              NaN  
DID_THE_HOUSEHOLD_ERN_INCOME_FRO           NaN  
OTHER_SPECIFY                              NaN  
HOW_MUCH_DID_THE_HOUSEHOLD_EARN_    60000000.0  
HOW_MUCH_DID_THE_HOUSEHOLDS_RECE     7000000.0  
TOTAL_INCOME                        60000000.0

In [68]:
# see percentage of INCOME_SOURCE
df14['INCOME_SOURCE'].value_counts(normalize=True)

INCOME_SOURCE
Transportation businesses         0.034483
Grain milling                     0.034483
Maize sales                       0.034483
Casual labour on farm (on-farm    0.034483
Fruits Sales                      0.034483
Brick making                      0.034483
Salaried employment               0.034483
Other income source (specify)     0.034483
Sugar cane                        0.034483
Casual labour off farm            0.034483
Cotton sales                      0.034483
selling firewood                  0.034483
Sorghum                           0.034483
Shop                              0.034483
Cassava                           0.034483
Selling charcoal                  0.034483
Salon                             0.034483
Soya beans sales                  0.034483
Sunflower sales                   0.034483
Vegetable                         0.034483
Grain trading                     0.034483
Other non-agricultural busines    0.034483
Tobacco Sales                     0.0344

In [71]:
df15.describe(include='all').transpose()

count unique               top  freq   
HH_ID                            11028.0    NaN               NaN   NaN  \
SERIAL_NOG                       11028.0    NaN               NaN   NaN   
ANIMAL_TYPE                        11028     12  Indigenous  cows   919   
DOES_THE_HOUSEHOLD_OWN_ANIMAL      11028      2                No  8889   
TOTAL_OWNED_BY_HOUSEHOLD          2139.0    NaN               NaN   NaN   
AVERAGE_VALUE_IF_SOLD_TODAY_ETB   2139.0    NaN               NaN   NaN   
DID_THE_HOUSEHOLD_SELL              2139      2                No  1600   
QUANTITY_SOLD1                     539.0    NaN               NaN   NaN   
M410_UNIT                            539      1            Number   539   
AVERAGE_SELLING_PRICE              539.0    NaN               NaN   NaN   

                                            mean               std       min   
HH_ID                            12073027.243743  174132911.655463  110104.0  \
SERIAL_NOG                                   6.5          3.452209       1.0   
ANIMAL_TYPE                                  NaN               NaN       NaN   
DOES_THE_HOUSEHOLD_OWN_ANIMAL                NaN               NaN       NaN   
TOTAL_OWNED_BY_HOUSEHOLD               6563.5741      89140.244972       0.0   
AVERAGE_VALUE_IF_SOLD_TODAY_ETB    433553.046751     930960.892787       0.0   
DID_THE_HOUSEHOLD_SELL                       NaN               NaN       NaN   
QUANTITY_SOLD1                       1140.048237      21726.309202       1.0   
M410_UNIT                                    NaN               NaN       NaN   
AVERAGE_SELLING_PRICE              217876.656772     422251.341733      10.0   

                                      25%       50%       75%           max  
HH_ID                            320018.0  560380.0  800003.0  3500269999.0  
SERIAL_NOG                           3.75       6.5      9.25          12.0  
ANIMAL_TYPE                           NaN       NaN       NaN           NaN  
DOES_THE_HOUSEHOLD_OWN_ANIMAL         NaN       NaN       NaN           NaN  
TOTAL_OWNED_BY_HOUSEHOLD              1.0       2.0       5.0     2500000.0  
AVERAGE_VALUE_IF_SOLD_TODAY_ETB   40000.0  150000.0  500000.0    22500000.0  
DID_THE_HOUSEHOLD_SELL                NaN       NaN       NaN           NaN  
QUANTITY_SOLD1                        1.5       3.0       5.0      500000.0  
M410_UNIT                             NaN       NaN       NaN           NaN  
AVERAGE_SELLING_PRICE             20000.0   80000.0  200000.0     4500000.0

In [72]:
df16.describe(include='all').transpose()

count unique             top freq   
HH_ID                             920.0    NaN             NaN  NaN  \
A1                                  920     11         Wankole  156   
A2                                  920    166       Nawanyago   34   
A3                                  920    406          bugobi   15   
A4                                  920     51          Buzaya  177   
A6                                  920      2          Female  530   
A8                                  920      2            Male  787   
A9                                  772      4  Household head  472   
NAME_OF_THE_GROUP                   920    451    Balitwegomba   19   
POSITION_IN_THE_GROUP               920      2          Member  667   
HOW_MANY_YEARS_HAVE_YOU_BEEN_A_M  920.0    NaN             NaN  NaN   
MILKING_COWS                        919      2              No  729   
IS_FODDER_A_MAJOR_CONSTRAINT_TO     919      2              No  607   
DID_YOU_BUY_FODDER_HAY_FROM_GR      919      2              No  775   
IF_YES_WHAT_TYPE_OF_FODDER_DID_Y    144      2   Other specify  113   
OTHER_TYPE_OF_FODDER                920     40                  801   
IF_YES_HOW_MUCH_DID_YOU_SPEND     144.0    NaN             NaN  NaN   
A5                                  920      8                  913   
A7                                  920      8                  913   
A4_1                                  7      2             Yes    6   
A9_OTHER                            920      1                  920   
A11                                 920      8                  913   
A12                                 920      7                  913   
_merge                                7      1     matched (3)    7   

                                             mean               std       min   
HH_ID                             12060774.104348  174125413.075476  110104.0  \
A1                                            NaN               NaN       NaN   
A2                                            NaN               NaN       NaN   
A3                                            NaN               NaN       NaN   
A4                                            NaN               NaN       NaN   
A6                                            NaN               NaN       NaN   
A8                                            NaN               NaN       NaN   
A9                                            NaN               NaN       NaN   
NAME_OF_THE_GROUP                             NaN               NaN       NaN   
POSITION_IN_THE_GROUP                         NaN               NaN       NaN   
HOW_MANY_YEARS_HAVE_YOU_BEEN_A_M             4.26          3.608111       0.0   
MILKING_COWS                                  NaN               NaN       NaN   
IS_FODDER_A_MAJOR_CONSTRAINT_TO               NaN               NaN       NaN   
DID_YOU_BUY_FODDER_HAY_FROM_GR                NaN               NaN       NaN   
IF_YES_WHAT_TYPE_OF_FODDER_DID_Y              NaN               NaN       NaN   
OTHER_TYPE_OF_FODDER                          NaN               NaN       NaN   
IF_YES_HOW_MUCH_DID_YOU_SPEND            267875.0     450066.088776    3000.0   
A5                                            NaN               NaN       NaN   
A7                                            NaN               NaN       NaN   
A4_1                                          NaN               NaN       NaN   
A9_OTHER                                      NaN               NaN       NaN   
A11                                           NaN               NaN       NaN   
A12                                           NaN               NaN       NaN   
_merge                                        NaN               NaN       NaN   

                                        25%       50%        75%           max  
HH_ID                             320023.25  560380.5  800003.25  3500269999.0  
A1                                      NaN       NaN        NaN      

In [73]:
# see percentage of IS_FODDER_A_MAJOR_CONSTRAINT_TO
df16['IS_FODDER_A_MAJOR_CONSTRAINT_TO'].value_counts(normalize=True)    

IS_FODDER_A_MAJOR_CONSTRAINT_TO
No     0.660501
Yes    0.339499
Name: proportion, dtype: float64

In [74]:
# see percentage of DID_YOU_BUY_FODDER_HAY_FROM_GR: Did you buy fodder in the last 12 months
df16['DID_YOU_BUY_FODDER_HAY_FROM_GR'].value_counts(normalize=True)

DID_YOU_BUY_FODDER_HAY_FROM_GR
No     0.843308
Yes    0.156692
Name: proportion, dtype: float64

In [75]:
df17.describe(include='all').transpose()

count unique                      top  freq             mean   
HH_ID    3676.0    NaN                      NaN   NaN  12073027.243743  \
ITEM_SN  3676.0    NaN                      NaN   NaN              2.5   
M4A12      3676      5  Artificial insemination   919              NaN   
M4A13      3676      2                       No  2875              NaN   
M4A14     801.0    NaN                      NaN   NaN    158455.580524   

                      std       min       25%       50%       75%   
HH_ID    174148705.307634  110104.0  320018.0  560380.0  800003.0  \
ITEM_SN          1.118186       1.0      1.75       2.5      3.25   
M4A12                 NaN       NaN       NaN       NaN       NaN   
M4A13                 NaN       NaN       NaN       NaN       NaN   
M4A14       351161.723563      60.0   30000.0   60000.0  150000.0   

                  max  
HH_ID    3500269999.0  
ITEM_SN           4.0  
M4A12             NaN  
M4A13             NaN  
M4A14       5000000.0

In [76]:
df18.describe(include='all').transpose()

count unique   
HH_ID                            198.0    NaN  \
SERIAL_NT                        198.0    NaN   
TYPE_OF_MILKING_COWS               198      2   
NUMBER_OF_MILKING_COWS           198.0    NaN   
CALVING_INTERVAL_YEAR            198.0    NaN   
AVERAGE_LACTATION_LENGTH_MONTHS  198.0    NaN   
TOTAL_AMOUNT_OF_MILK_PRODUCED_I  198.0    NaN   
WHO_MILKS_THE_COWS                 198     10   
WHO_SELLS_THE_MILK                 198      9   
WHO_USES_INCOME_FROM_MILK_SALE     198     11   
WHO_CONSUMES_MILK1                 198     11   
ADD_OTHER_MILKING_COW_TYPE         198      2   

                                                                               top   
HH_ID                                                                          NaN  \
SERIAL_NT                                                                      NaN   
TYPE_OF_MILKING_COWS                                                         Local   
NUMBER_OF_MILKING_COWS                                                         NaN   
CALVING_INTERVAL_YEAR                                                          NaN   
AVERAGE_LACTATION_LENGTH_MONTHS                                                NaN   
TOTAL_AMOUNT_OF_MILK_PRODUCED_I                                                NaN   
WHO_MILKS_THE_COWS                                                            Head   
WHO_SELLS_THE_MILK                                                            Head   
WHO_USES_INCOME_FROM_MILK_SALE   Head , spouse and other male +female family me...   
WHO_CONSUMES_MILK1               Head , spouse and other male +female family me...   
ADD_OTHER_MILKING_COW_TYPE                                                      No   

                                freq           mean            std       min   
HH_ID                            NaN  572883.732323  387317.637941  120001.0  \
SERIAL_NT                        NaN       1.040404       0.197404       1.0   
TYPE_OF_MILKING_COWS             100            NaN            NaN       NaN   
NUMBER_OF_MILKING_COWS           NaN       3.141414      25.518784       0.0   
CALVING_INTERVAL_YEAR            NaN      11.906566      143.29905       0.2   
AVERAGE_LACTATION_LENGTH_MONTHS  NaN       7.165505       3.172099       0.0   
TOTAL_AMOUNT_OF_MILK_PRODUCED_I  NaN     1506.39899    5701.980202       0.0   
WHO_MILKS_THE_COWS               118            NaN            NaN       NaN   
WHO_SELLS_THE_MILK                94            NaN            NaN       NaN   
WHO_USES_INCOME_FROM_MILK_SALE    98            NaN            NaN       NaN   
WHO_CONSUMES_MILK1               155            NaN            NaN       NaN   
ADD_OTHER_MILKING_COW_TYPE       190            NaN            NaN       NaN   

                                       25%       50%       75%        max  
HH_ID                            220656.25  460489.5  890034.5  1390144.0  
SERIAL_NT                              1.0       1.0       1.0        2.0  
TYPE_OF_MILKING_COWS                   NaN       NaN       NaN        NaN  
NUMBER_OF_MILKING_COWS                 1.0       1.0       1.0      360.0  
CALVING_INTERVAL_YEAR                  1.0       1.0       2.0     2018.0  
AVERAGE_LACTATION_LENGTH_MONTHS        6.0       8.0       9.0       15.0  
TOTAL_AMOUNT_OF_MILK_PRODUCED_I      240.0     540.0    1200.0    72000.0  
WHO_MILKS_THE_COWS                     NaN       NaN       NaN        NaN  
WHO_SELLS_THE_MILK                     NaN       NaN       NaN        NaN  
WHO_USES_INCOME_FROM_MILK_SALE         NaN       NaN       NaN        NaN  
WHO_CONSUMES_MILK1                     NaN       NaN       NaN        NaN  
ADD_OTHER_MILKING_COW_TYPE             NaN       NaN       NaN        NaN

In [77]:
df19.describe(include='all').transpose()

count unique     top  freq   
HH_ID                             7344.0    NaN     NaN   NaN  \
SERIAL_NUMBER                     7344.0    NaN     NaN   NaN   
PRODUCT                             7344      8    Eggs   918   
DID_THE_HOUSEHOLD_PRODUCE_PRODUC    7344      2      No  6870   
UNIT1                                474      4  Number   177   
AMOUNT_PRODUCED                    474.0    NaN     NaN   NaN   
QUANTITY_CONSUMED_AT_HOME          474.0    NaN     NaN   NaN   
QUANTITY_SOLD                      474.0    NaN     NaN   NaN   
AVERAGE_PRICE_UGX                  474.0    NaN     NaN   NaN   
TOTAL_VALUE_SOLD_8B15X_8B16          0.0    NaN     NaN   NaN   
QUANTITY_USED_FOR_OTHER_PURPOSE    474.0    NaN     NaN   NaN   
DID_YOU_BUY_PRODUCT                  474      2      No   448   
IF_YES_AMOUNT_PURCHASED             26.0    NaN     NaN   NaN   
UNIT2                                 26      3  Number    15   
VALUE_OF_PURCHASED                 474.0    NaN     NaN   NaN   

                                             mean               std       min   
HH_ID                             12085840.964052  174231259.463456  110104.0  \
SERIAL_NUMBER                                 4.5          2.291444       1.0   
PRODUCT                                       NaN               NaN       NaN   
DID_THE_HOUSEHOLD_PRODUCE_PRODUC              NaN               NaN       NaN   
UNIT1                                         NaN               NaN       NaN   
AMOUNT_PRODUCED                        464.908228       1151.699415       0.0   
QUANTITY_CONSUMED_AT_HOME              179.797468        654.618025      -5.0   
QUANTITY_SOLD                          220.987342        735.041499       0.0   
AVERAGE_PRICE_UGX                    81841.742616     733765.867134       0.0   
TOTAL_VALUE_SOLD_8B15X_8B16                   NaN               NaN       NaN   
QUANTITY_USED_FOR_OTHER_PURPOSE        280.244726       1525.645966       0.0   
DID_YOU_BUY_PRODUCT                           NaN               NaN       NaN   
IF_YES_AMOUNT_PURCHASED                 17.884615         35.680053       0.0   
UNIT2                                         NaN               NaN       NaN   
VALUE_OF_PURCHASED                    2747.529536      11153.636742       0.0   

                                       25%       50%       75%           max  
HH_ID                             320025.0  560380.5  800003.0  3500269999.0  
SERIAL_NUMBER                         2.75       4.5      6.25           8.0  
PRODUCT                                NaN       NaN       NaN           NaN  
DID_THE_HOUSEHOLD_PRODUCE_PRODUC       NaN       NaN       NaN           NaN  
UNIT1                                  NaN       NaN       NaN           NaN  
AMOUNT_PRODUCED                       10.0      40.0    403.75       10000.0  
QUANTITY_CONSUMED_AT_HOME              3.0      12.0     120.0       10000.0  
QUANTITY_SOLD                          0.0       0.0      23.0        9240.0  
AVERAGE_PRICE_UGX                      0.0       0.0    1000.0    14960000.0  
TOTAL_VALUE_SOLD_8B15X_8B16            NaN       NaN       NaN           NaN  
QUANTITY_USED_FOR_OTHER_PURPOSE        0.0       0.0       0.0        9999.0  
DID_YOU_BUY_PRODUCT                    NaN       NaN       NaN           NaN  
IF_YES_AMOUNT_PURCHASED                2.0       8.0      15.0         180.0  
UNIT2                                  NaN       NaN       NaN           NaN  
VALUE_OF_PURCHASED                     0.0       0.0       0.0      180000.0

In [78]:
df20.describe(include='all').transpose()

count unique             top   freq   
HH_ID                             920.0    NaN             NaN    NaN  \
A1                                  920     11         Wankole    156   
A2                                  920    166         Luzinga     34   
A3                                  920    406          bugobi     15   
A4                                  920     51          Buzaya    177   
A6                                  920      2          Female    530   
A8                                  920      2            Male    787   
A9                                  772      4  Household head    472   
NAME_OF_THE_GROUP                   920    451    Balitwegomba     19   
POSITION_IN_THE_GROUP               920      2          Member    667   
HOW_MANY_YEARS_HAVE_YOU_BEEN_A_M  920.0    NaN             NaN    NaN   
A1_9A                               918     84  Not Applicable     76   
A1_9B                             918.0   66.0            20.0   81.0   
A2_9A                               918      3             Yes    643   
A2_9B                               918      4             Yes    784   
A3_9A                               918      3              No    518   
A3_9B                               918      4              No    536   
A4_9                              918.0    NaN             NaN    NaN   
A5_9                                918      4             Yes    625   
A6_9A                               918      4             Yes    620   
A6_9B                               918      4             Yes    746   
A7_9A                               918      4             Yes    404   
A7_9B                               918      4             Yes    463   
HG09                                918      3             Yes    543   
HG10                              918.0   21.0             3.0  154.0   

                                             mean               std       min   
HH_ID                             12060774.104348  174125413.075476  110104.0  \
A1                                            NaN               NaN       NaN   
A2                                            NaN               NaN       NaN   
A3                                            NaN               NaN       NaN   
A4                                            NaN               NaN       NaN   
A6                                            NaN               NaN       NaN   
A8                                            NaN               NaN       NaN   
A9                                            NaN               NaN       NaN   
NAME_OF_THE_GROUP                             NaN               NaN       NaN   
POSITION_IN_THE_GROUP                         NaN               NaN       NaN   
HOW_MANY_YEARS_HAVE_YOU_BEEN_A_M             4.26          3.608111       0.0   
A1_9A                                         NaN               NaN       NaN   
A1_9B                                         NaN               NaN       NaN   
A2_9A                                         NaN               NaN       NaN   
A2_9B                                         NaN               NaN       NaN   
A3_9A                                         NaN               NaN       NaN   
A3_9B                                         NaN               NaN       NaN   
A4_9                                    68.979303         91.583469     -99.0   
A5_9                                          NaN               NaN       NaN   
A6_9A                                         NaN               NaN       NaN   
A6_9B                                         NaN               NaN       NaN   
A7_9A                                         NaN               NaN       NaN   
A7_9B                                         NaN               NaN       NaN   
HG09                                          NaN               NaN       NaN   
HG10                                          NaN               NaN       NaN   

                            

In [80]:
dfv20.variable_labels()

{'HH_ID': 'Household Id',
 'A1': 'A2: Subcounty',
 'A2': 'A3: Parish',
 'A3': 'A4: Village',
 'A4': 'A1 County',
 'A6': 'A6: Sex of the respondent',
 'A8': 'A8: Sex of household head',
 'A9': 'A9 Relationship of Respondent to Household Head',
 'NAME_OF_THE_GROUP': 'Name of the group',
 'POSITION_IN_THE_GROUP': 'Position in the group',
 'HOW_MANY_YEARS_HAVE_YOU_BEEN_A_M': 'How many years have you been a member to this group',
 'A1_9A': '5A1: How many years have you [PRIMARY MALE] been living in this village? (in yea',
 'A1_9B': '5A1: How many years have you [PRIMARY FEMALE] been living in this village? (in y',
 'A2_9A': '5A2_a: In the last 12 months, did you [PRIMARY MALE] have a need for advice or i',
 'A2_9B': '5A2_b: In the last 12 months, did you [PRIMARY FEMALE] have a need for advice or',
 'A3_9A': '5A3_a: In the last 12 months, did you [PRIMARY MALE] receive any advice or infor',
 'A3_9B': '5A3_b: In the last 12 months, did you [PRIMARY FEMALE] receive any advice or inf',
 'A4_9'

In [83]:
df21.describe(include='all').transpose()

count unique                                            top   
HH_ID         20196.0    NaN                                            NaN  \
SR_NO         20196.0    NaN                                            NaN   
FOOD            20196     23                                        CEREALS   
DESCRIPTION     20196     32  Red palm oil, palm nut or palm nut pulp sauce   
CHILD_24DAYS    10999      3                                             No   
CHILD_7DAYS     10999      3                                            Yes   
MOTHER_24HRS    20075      3                                             No   
MOTHER_7DAYS    20075      3                                            Yes   

               freq             mean             std       min       25%   
HH_ID           NaN  12085840.964052  174223710.2852  110104.0  320025.0  \
SR_NO           NaN             11.5        6.344446       1.0       6.0   
FOOD            918              NaN             NaN       NaN       NaN   
DESCRIPTION     917              NaN             NaN       NaN       NaN   
CHILD_24DAYS   6667              NaN             NaN       NaN       NaN   
CHILD_7DAYS    5978              NaN             NaN       NaN       NaN   
MOTHER_24HRS  12117              NaN             NaN       NaN       NaN   
MOTHER_7DAYS  10751              NaN             NaN       NaN       NaN   

                   50%       75%           max  
HH_ID         560380.5  800003.0  3500269999.0  
SR_NO             11.5      17.0          22.0  
FOOD               NaN       NaN           NaN  
DESCRIPTION        NaN       NaN           NaN  
CHILD_24DAYS       NaN       NaN           NaN  
CHILD_7DAYS        NaN       NaN           NaN  
MOTHER_24HRS       NaN       NaN           NaN  
MOTHER_7DAYS       NaN       NaN           NaN

In [84]:
dfv21.variable_labels()

{'HH_ID': 'Household Id',
 'SR_NO': 'Sr.No.',
 'FOOD': 'Food',
 'DESCRIPTION': 'Description',
 'CHILD_24DAYS': 'Did Child/children (up to 59 months) eat the [food] yesterday (day & night)?',
 'CHILD_7DAYS': 'Did Child/children (up to 59 months) eat the [food] in 7 days?',
 'MOTHER_24HRS': 'Did Mother eat the [food] yesterday (day & night)?',
 'MOTHER_7DAYS': 'Did mother eat the [food] in 7 days?'}

In [85]:
df22.describe(include='all').transpose()

count unique   
HH_ID                             920.0    NaN  \
A1                                  920     11   
A2                                  920    166   
A3                                  920    406   
A4                                  920     51   
A6                                  920      2   
A8                                  920      2   
A9                                  772      4   
NAME_OF_THE_GROUP                   920    451   
POSITION_IN_THE_GROUP               920      2   
HOW_MANY_YEARS_HAVE_YOU_BEEN_A_M  920.0    NaN   
CHILDREN_BELOW_59_MONTHS            918      2   
FEMALE_MEMBERS                        0      0   
foodadequacy                        918      3   
housingadeqcy                       918      3   
clothingadqcy                       918      4   
healthcareadqcy                     918      3   
schoolsadqcy                        918      4   
HOW_IS_THE_CONSUMPTION_HABIT_IN     911      4   
DID_YOU_OR_ANYONE_IN_YOUR_HOUSEH    911      2   

                                                                                top   
HH_ID                                                                           NaN  \
A1                                                                          Wankole   
A2                                                                          Luzinga   
A3                                                                           bugobi   
A4                                                                           Buzaya   
A6                                                                           Female   
A8                                                                             Male   
A9                                                                   Household head   
NAME_OF_THE_GROUP                                                      Balitwegomba   
POSITION_IN_THE_GROUP                                                        Member   
HOW_MANY_YEARS_HAVE_YOU_BEEN_A_M                                                NaN   
CHILDREN_BELOW_59_MONTHS                                                        Yes   
FEMALE_MEMBERS                                                                  NaN   
foodadequacy                      It was less than adequate for your family's needs   
housingadeqcy                         It was just adequate for your familys's needs   
clothingadqcy                     It was less than adequate for your family's needs   
healthcareadqcy                   It was less than adequate for your family's needs   
schoolsadqcy                      It was less than adequate for your family's needs   
HOW_IS_THE_CONSUMPTION_HABIT_IN   All household members including children eat s...   
DID_YOU_OR_ANYONE_IN_YOUR_HOUSEH                                                 No   

                                 freq             mean               std   
HH_ID                             NaN  12060774.104348  174125413.075476  \
A1                                156              NaN               NaN   
A2                                 34              NaN               NaN   
A3                                 15              NaN               NaN   
A4                                177              NaN               NaN   
A6                                530              NaN               NaN   
A8                                787              NaN               NaN   
A9                                472              NaN               NaN   
NAME_OF_THE_GROUP                  19              NaN               NaN   
POSITION_IN_THE_GROUP             667              NaN               NaN   
HOW_MANY_YEARS_HAVE_YOU_BEEN_A_M  NaN             4.26          3.608111   
CHILDREN_BELOW_59_MONTHS          503              NaN               NaN   
FEMALE_MEMBERS                    NaN              NaN               NaN   
foodadequacy                      451              NaN               NaN   
housingadeqcy  

In [86]:
dfv22.variable_labels()

{'HH_ID': 'Household Id',
 'A1': 'A2: Subcounty',
 'A2': 'A3: Parish',
 'A3': 'A4: Village',
 'A4': 'A1 County',
 'A6': 'A6: Sex of the respondent',
 'A8': 'A8: Sex of household head',
 'A9': 'A9 Relationship of Respondent to Household Head',
 'NAME_OF_THE_GROUP': 'Name of the group',
 'POSITION_IN_THE_GROUP': 'Position in the group',
 'HOW_MANY_YEARS_HAVE_YOU_BEEN_A_M': 'How many years have you been a member to this group',
 'CHILDREN_BELOW_59_MONTHS': 'Does this household have any children below 59 months',
 'FEMALE_MEMBERS': 'Does this household have any female members',
 'foodadequacy': 'What is your own assessment of the adequacy of your family?s food consumption ov',
 'housingadeqcy': 'What is your own assessment of the adequacy of your family?s housing over the pa',
 'clothingadqcy': 'What is your own assessment of the adequacy of your family?s clothing over the p',
 'healthcareadqcy': 'What is your own assessment of the adequacy of the health care your family gets ',
 'schoolsa